### PR TITLE
libgit2: Enable tests, multi-outputs

### PR DESCRIPTION
**Pull request by roberth**
## Description of changes

- Enable tests
- Enable multiple outputs

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/libgit2/default.nix [hunk 1]
```diff
@@ -46,7 +46,23 @@ stdenv.mkDerivation rec {
...
+    # failed to set permissions on ...: Operation not permitted
+    testArgs+=(-xrepo::init::extended_1)
+    testArgs+=(-xrepo::template::extended_with_template_and_shared_mode)
+
+    ( set -x
+      ./libgit2_tests ''${testArgs[@]}
+    )
+  '';
+
+  outputs = ["lib" "dev" "out"];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgit2/default.nix:65]: Usually we place outputs near the top after version and before src. Also some extra spaces inside the square brackets does not make it so tight and cramped.

File: pkgs/development/libraries/libgit2/default.nix [hunk 2]
```diff
@@ -46,7 +46,23 @@ stdenv.mkDerivation rec {
...
+    # slow
+    testArgs+=(-xclone::nonetwork::bad_urls)
+
+    # failed to set permissions on ...: Operation not permitted
+    testArgs+=(-xrepo::init::extended_1)
+    testArgs+=(-xrepo::template::extended_with_template_and_shared_mode)
+
+    ( set -x
+      ./libgit2_tests ''${testArgs[@]}
+    )
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgit2/default.nix:62]:
```suggestion
    (
      set -x
      ./libgit2_tests ''${testArgs[@]}
    )
```
I am used to shfmt google style which I am pretty sure would suggest this

**Review by SuperSandro2000 — APPROVED** [review]
with that LGTM, feel free to merge then

**Comment by roberth** [comment]
Thank you for reviewing as maintainer of this package @SuperSandro2000!

**Comment by trofi** [comment]
This change broke `libgit2_1_3_0` build and a few of it's reverse dependencies. Hydra sees the failure as https://hydra.nixos.org/job/nixpkgs/trunk/libgit2_1_3_0.x86_64-linux

```
Running phase: checkPhase
++ ./libgit2_tests -v -xonline -xclone::nonetwork::bad_urls -xrepo::init::extended_1 -xrepo::template::extended_with_template_and_shared_mode
/nix/store/yq9qag5ak3w15nhcqplnhxpny6295l36-stdenv-linux/setup: line 1593: ./libgit2_tests: No such file or directory
```

**Comment by roberth** [comment]
@trofi please review
- https://github.com/NixOS/nixpkgs/pull/289802

**Pull request merged**